plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorTest.java

```diff
@@ -0,0 +1,234 @@
+    }
+
+    @Override
+    protected String createSchemaSql(String schemaName)
+    {
+        return "CREATE SCHEMA IF NOT EXISTS " + schemaName + " WITH (location = 's3://" + bucketName + "/" + schemaName + "')";
+    }
+
+    @Override
+    protected Long calculateFileSystemFileSize(String filePath)
+    {
+        return this.hiveMinioDataLake.getS3Client()
+                .getObject(bucketName, filePath.replace("s3://" + bucketName + "/", ""))
+                .getObjectMetadata()
+                .getContentLength();
+    }
+
+    @Override
+    protected BufferedInputStream createDataFileStreamForFileSystem(String filePath)
+    {
+        return new BufferedInputStream(this.hiveMinioDataLake.getS3Client()
+                .getObject(bucketName, filePath.replace("s3://" + bucketName + "/", "")).getObjectContent());
+    }
+
+    @Override
+    protected List<String> getAllDataFilesFromTableDirectory(String tableName)
+    {
+        String tableLocation = getSession().getSchema().orElseThrow() + "/" + tableName + "/data";
+        return this.hiveMinioDataLake.getS3Client().listObjects(bucketName, tableLocation)
+                .getObjectSummaries()
+                .stream()
+                .map(object -> "s3://" + bucketName + "/" + object.getKey())
+                .filter(object -> !object.endsWith(".crc"))
+                .collect(toImmutableList());
+    }
+
+    @Test
+    public void testRepartitionDataOnCtasPartitionComplex()
+    {
+        // complex; would exceed 100 open writers limit in IcebergPageSink without write repartitioning
+        testRepartitionData(getSession(), "tpch.tiny.orders", Optional.of("orderkey"), Optional.of(1500), true, "'bucket(custkey, 4)', 'truncate(comment, 1)'", 111);
+    }
+
+    @Test
+    public void testRepartitionDataOnInsertPartitionComplex()
+    {
+        // complex; would exceed 100 open writers limit in IcebergPageSink without write repartitioning
+        testRepartitionData(getSession(), "tpch.tiny.orders", Optional.of("orderkey"), Optional.of(1500), false, "'bucket(custkey, 4)', 'truncate(comment, 1)'", 111);
+    }
+
+    @Test
+    public void testRepartitionDataOnCtasPartitionWithSameColumnMultipleTimes()
+    {
+        testRepartitionData(getSession(), "tpch.tiny.orders", Optional.of("orderkey"), Optional.of(300), true, "'truncate(comment, 1)', 'orderstatus', 'bucket(comment, 2)'", 97);
+    }
+
+    @Test
+    public void testRepartitionDataOnInsertPartitionWithSameColumnMultipleTimes()
+    {
+        testRepartitionData(getSession(), "tpch.tiny.orders", Optional.of("orderkey"), Optional.of(300), false, "'truncate(comment, 1)', 'orderstatus', 'bucket(comment, 2)'", 97);
+    }
+
+    /**
+     * DataProvider overload to exclude 2 test cases from generic tests:
+     * - {@link BaseIcebergConnectorTest#testRepartitionDataOnCtas(Session, String, int)}
+     * - {@link BaseIcebergConnectorTest#testRepartitionDataOnInsert(Session, String, int)}
+     * <p>
```

Comment:
If that's the intention, this should be imeplemented as

```
repartitioningDataProvider() {

  defaultTestData = super.repartitioningDataProvider();
  // do filter and return
```

then, the rest of commentary would go inside the method, adorning the filtering conditions.

Reply:
BTW, awesome commentary

Reply:
Good point. Overseen that. It we will ovoid loosing test cases added later on in base test.
Modified to filter out unwanted test cases.